src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,518 @@
+     *
+     * @param typeOfTask Guaranteed to be either COMMAND_TODO, COMMAND_DEADLINE, or COMMAND_EVENT
+     * @param params String containing description and other info of the task
+     */
+    public static void addTask(String typeOfTask, String params) {
+        final String[] descriptionAndInfo;
+        try {
+            descriptionAndInfo = splitParamsIntoDescriptionAndInfo(params);
+            final String description = descriptionAndInfo[0];
+            final String info = descriptionAndInfo[1];
+            Task task;
+            switch (typeOfTask){
+            case COMMAND_TODO:
+                task = new Todo(description);
+                break;
+            case COMMAND_DEADLINE:
+                String dateDeadline = extractDate(COMMAND_DEADLINE_PREFIX, COMMAND_DEADLINE, info);
+                task = new Deadline(description, dateDeadline);
+                break;
+            case COMMAND_EVENT:
+                String dateEvent = extractDate(COMMAND_EVENT_PREFIX, COMMAND_EVENT, info);
+                task = new Event(description, dateEvent);
+                break;
+            default:
+                //Since typeofTask is guaranteed to be either of the 3 above, code should not be reaching this point
+                throw new InvalidCommandFormatException("Error occurred, please check command format!");
+            }
+
+            tasks.add(task);
+            appendNewTaskToFile(task);
+            showMessageFramedWithDivider("Added to list: " , task.toString(),
+                    "Current number of tasks: " + tasks.size());
+        } catch (InvalidCommandFormatException e) {
+            //Print error for any invalid command format exceptions caught
+            showMessageFramedWithDivider(e.toString());
+        }
+    }
+
+    /**
+     * Deletes a specific task
+     * If task number is not provided or invalid, prints an error
+     *
+     * @param params String in the format "delete X", where X is supposed to be the task number
+     */
+    public static void deleteTask(String params) {
+        if (tasks.isEmpty()) {
+            //error if user does not have any tasks to be deleted
+            showMessageFramedWithDivider(MESSAGE_NO_TASKS_YET);
+        } else if (params.equals(EMPTY)) {
+            //error if user inputs only "delete" with no number behind
+            showMessageFramedWithDivider(MESSAGE_ERROR_INVALID_COMMAND_DELETE_FORMAT);
+        } else {
+            try {
+                int taskNum = Integer.parseInt(params);
+                if (taskNum > tasks.size() || taskNum < 1) {
+                    //error if user inputs a task number that does not exist
+                    showMessageFramedWithDivider("Please input a valid task number from 1 to " + tasks.size() + "!");
+                } else {
+                    final String deletedTaskString = tasks.get(taskNum-1).toString();
+                    tasks.remove(taskNum-1);
```

Comment:
As mentioned before.
```suggestion
                    tasks.remove(taskNum - 1);
```